coral/src/app/components/documentation/DocumentationEditor.tsx

```diff
@@ -0,0 +1,120 @@
+import { Light as SyntaxHighlighter } from "react-syntax-highlighter";
+import { a11yLight } from "react-syntax-highlighter/dist/esm/styles/hljs";
+import {
+  Box,
+  Button,
+  SegmentedControl,
+  SegmentedControlGroup,
+  Typography,
+} from "@aivenio/aquarium";
+import classes from "src/app/components/documentation/documentation-editor.module.css";
+import { useState } from "react";
+import { DocumentationView } from "src/app/components/documentation/DocumentationView";
+
+type DocumentationEditorProps = {
+  documentation?: string;
+  save: (documentation: string) => void;
+  cancel: () => void;
+  isSaving?: boolean;
+};
+
+type ViewMode = "edit" | "preview";
+function DocumentationEditor({
+  documentation,
+  save,
+  cancel,
+  isSaving,
+}: DocumentationEditorProps) {
+  const [viewMode, setViewMode] = useState<ViewMode>("edit");
+  const [text, setText] = useState(documentation || "");
+
+  function saveDocumentation() {
+    // trim to remove unnecessary whitespace at end/start
+    const newDocumentation = text.trim();
+
+    // existing `documentation` needs to be trimmed to compare, since
+    // there we don't know if the value saved before migrating to React
+    // contains spaces at beginning / end
+    if (documentation && newDocumentation === documentation.trim()) return;
+    save(newDocumentation);
+  }
+
+  return (
+    <Box.Flex flexDirection={"column"} rowGap={"3"}>
+      <Box.Flex
+        alignSelf={"start"}
+        component={"section"}
+        aria-label={"Switch between edit and preview mode"}
+      >
+        <SegmentedControlGroup
+          onChange={(value: ViewMode) => setViewMode(value)}
+          value={viewMode}
+        >
+          <SegmentedControl aria-pressed={viewMode === "edit"} value={"edit"}>
+            Edit
+          </SegmentedControl>
+          <SegmentedControl
+            aria-pressed={viewMode === "preview"}
+            value={"preview"}
+          >
+            Preview
+          </SegmentedControl>
+        </SegmentedControlGroup>
+      </Box.Flex>
+
+      {viewMode === "edit" && (
+        <div className={classes.markdownEditor}>
+          <Box
+            borderColor={"grey-20"}
+            borderWidth={"1px"}
+            borderRadius={"2px"}
+            marginBottom={"2"}
+            className={classes.markdownSyntaxHighlight}
+          >
+            <label
+              className={"visually-hidden"}
+              htmlFor={"markdown-editor-textarea"}
+            >
+              Markdown editor
+            </label>
+            <textarea
```

Comment:
It took me a minute to understand this was overlaying two input components on top of each other and styling them to look like one. Was this done for accessibility purposes? Is this something that is done often? Candid question, because it is not something I would have thought of at all haha!

At any rate, I think a little comment to explain the approach would be nice ^^

Reply:
It was the easiest way to show syntax highlights markdown with a simple text area (which provides all accessibility we need).

Not sure if it's done often, as far as I've seen most people use something like code-mirror etc. But I think it's an unnecessary dependency and load for the simple use case we're having. The markdown editor libs I found that are a bit more light are usually not really good in accessibility (even the one I use in my side project 😭 ). 

I tested it on Chrome/Firefox and even Safari to make sure everything works like expected. 

Will add comments to it!

Reply:
Oh, I think I also completely blanked on the fact that `SyntaxHighlighter` is not an input. I think it's good then ^^